client/src/www/app/platform.ts

```diff
@@ -14,22 +14,20 @@
 
 import {Clipboard} from './clipboard';
 import {EnvironmentVariables} from './environment';
-import {TunnelFactory} from './tunnel';
+import {OutlineServerRepository} from './outline_server_repository';
 import {Updater} from './updater';
 import {UrlInterceptor} from './url_interceptor';
 import {VpnInstaller} from './vpn_installer';
+import {EventQueue} from '../model/events';
 import {OutlineErrorReporter} from '../shared/error_reporter';
 
 // Provides platform-specific dependencies.
 // TODO: Remove one of hasDeviceSupport and getServerFactory; they're almost the same
 //       thing and currently hasDeviceSupport is only used to populate the server list when running
 //       in demo mode.
 export interface OutlinePlatform {
-  // Returns true iff the system has support for proxying. When this returns false, the UI should
-  // assume it's running in demo mode, e.g. Electron on macOS.
-  hasDeviceSupport(): boolean;
-
-  getTunnelFactory(): TunnelFactory;
+  // Creates the OutlineServerRepository for this platform. Returns undefined if the platform is not supported.
+  newServerRepo(eventQueue: EventQueue): OutlineServerRepository | undefined;
```

Comment:
I'm going back and forth with the design.
After playing with it for a bit, I think having explicit wiring versus messing with globals is better for now. At least all the wiring is a bit more encapsulated is in cordova and electron main now.
Once we have a way to pick implementations at build time, we can revisit that.

Reply:
I like this latest snapshot a bit better too. Do we track the work to pick implementation at build time anywhere? Maybe a simple TODO comment will do for now.